lib/tasks/withdraw_and_redirect_section.rake

```diff
@@ -1,22 +1,29 @@
 require "csv"
 
-desc "Withdraw and redirect section, e.g withdraw_and_redirect_section['guidance/manual/section,/redirect/blah,true']"
+desc "Withdraw and redirect section. WARNING: discard_draft will destroy drafts, usage: withdraw_and_redirect_section['guidance/manual/section,/redirect/blah,false']"
 task :withdraw_and_redirect_section, %i[section_path redirect discard_draft] => :environment do |_, args|
   discard_draft = args.fetch(:discard_draft) == "true"
-  redirect_section(args.fetch(:section_path), args.fetch(:redirect), discard_draft)
+  redirect_section = WithdrawAndRedirectSection.new(
+    user: User.gds_editor,
+    section_path:,
+    redirect:,
+    discard_draft:,
+  )
+  redirect_section.execute
   puts "Section withdrawn and redirected to #{base_path}"
 end
 
-desc "Withdraw and redirect sections to their manual base path, e.g bulk_withdraw_and_redirect_section_to_manual['lib/tasks/data/withdraw_sections.csv']"
-task :bulk_withdraw_and_redirect_section_to_manual, %i[csv_path] => :environment do |_, args|
+desc "Just redirect sections to their manual base path, e.g bulk_redirect_section_to_manual['lib/tasks/data/withdraw_sections.csv']"
+task :bulk_redirect_section_to_manual, %i[csv_path] => :environment do |_, args|
   section_count = CSV.read(args.fetch(:csv_path), headers: true).length
   puts "Withdrawing and unpublishing #{section_count} sections..."
 
   failures = {}
   CSV.foreach(args.fetch(:csv_path), headers: true) do |row|
     section_path = row["section_path"]
     base_path = section_path.split("/")[0..1].join("/")
-    redirect_section(section_path, "/#{base_path}", true)
+    section_edition = SectionEdition.where(slug: section_path).first
+    Publishing::RedirectAdapter.redirect_section(section_edition, to: "/#{base_path}")
```

Comment:
Should this be withdrawing the sections after all, now that we know the manual page will show archived sections as long as the manual is in the withdrawn state?

Reply:
Agreed that this will put the sections into the same state as 900ish other sections which have a redirect in place on Publishing API but are not in the `archived` state in the Manuals Publisher database. We can look at fixing them separately another time.